test/jellyfish_web/integration/server_socket_test.exs

```diff
@@ -107,41 +106,47 @@ defmodule JellyfishWeb.Integration.ServerSocketTest do
 
     {:ok, peer_ws} = WS.start_link("ws://127.0.0.1:#{@port}/socket/peer/websocket")
 
-    auth_request = auth_request(peer_token)
+    auth_request = peer_auth_request(peer_token)
 
     :ok = WS.send_frame(peer_ws, auth_request)
 
-    assert_receive %{
-      "data" => %{"id" => ^peer_id, "type" => "peerConnected"},
-      "type" => "controlMessage"
-    }
+    assert_receive %{"data" => %{"type" => "authenticated"}, "type" => "controlMessage"}
 
     conn = delete(conn, ~p"/room/#{room_id}/")
     response(conn, :no_content)
 
-    assert_receive %{
-      "data" => %{"id" => ^peer_id, "type" => "peerDisconnected"},
-      "type" => "controlMessage"
-    }
+    receive do
+      msg when is_binary(msg) ->
+        case ControlMessage.decode(msg) do
+          %ControlMessage{
+            content: {:peerDisconnected, %PeerNotification{peerId: ^peer_id, roomId: ^room_id}}
+          } ->
+            assert true
+
+          _other ->
+            nil
+        end
+    end
```

Comment:
Is there a specific reason why this `receive` block is different than the one from lines 85-90?